// @adminjs/design-system/build/molecules/drawer/drawer.stories.js
import React from 'react';
import { Box, Button, Header, Icon, Text } from "../../index.js";
import { FullFeaturedFormGroup } from "../form-group/form-group.stories.js";
import { Drawer, DrawerContent, DrawerFooter } from "./index.js";
export const Default = {
  render: props => /*#__PURE__*/React.createElement(Drawer, props, /*#__PURE__*/React.createElement(DrawerContent, null, /*#__PURE__*/React.createElement(Header.H3, null, /*#__PURE__*/React.createElement(Button, {
    size: "icon",
    rounded: true,
    mr: "lg"
  }, /*#__PURE__*/React.createElement(Icon, {
    icon: "ChevronRight"
  })), "Edit"), /*#__PURE__*/React.createElement(Box, {
    my: "x3",
    p: 0
  }, /*#__PURE__*/React.createElement(Button, null, /*#__PURE__*/React.createElement(Icon, {
    icon: "Info"
  }), "Info"), /*#__PURE__*/React.createElement(Button, {
    ml: "lg"
  }, /*#__PURE__*/React.createElement(Icon, {
    icon: "Trash2"
  }), "Delete")), /*#__PURE__*/React.createElement(Header.H4, null, "Primary Fields"), /*#__PURE__*/React.createElement(Text, null, "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin nibh augue, suscipit a, scelerisque sed, lacinia in, mi. Cras vel lorem. Etiam pellentesque aliquet tellus. Phasellus pharetra nulla ac diam. Quisque semper justo at risus. Donec venenatis, turpis vel hendrerit interdum, dui ligula ultricies purus, sed"), /*#__PURE__*/React.createElement(FullFeaturedFormGroup, null)), /*#__PURE__*/React.createElement(DrawerFooter, null, /*#__PURE__*/React.createElement(Button, {
    variant: "contained"
  }, "Save")))
};
const meta = {
  title: 'DesignSystem/Molecules/Drawer',
  component: Drawer,
  args: {
    isHidden: false
  },
  argTypes: {
    isHidden: {
      control: {
        type: 'boolean'
      }
    }
  }
};
export default meta;
