// @adminjs/design-system/build/atoms/radio/radio.stories.js
import React from 'react';
import { Box, Label } from "../index.js";
import { Radio } from "./index.js";
export const Default = () => /*#__PURE__*/React.createElement(Box, {
  flex: true
}, /*#__PURE__*/React.createElement(Box, {
  flexDirection: "column",
  marginRight: 15
}, /*#__PURE__*/React.createElement(Radio, {
  id: "checkbox1",
  checked: true
}), /*#__PURE__*/React.createElement(Label, {
  inline: true,
  htmlFor: "checkbox1",
  ml: "default"
}, "Checked")), /*#__PURE__*/React.createElement(Box, {
  flexDirection: "column",
  marginRight: 15
}, /*#__PURE__*/React.createElement(Radio, {
  id: "checkbox2",
  checked: false
}), /*#__PURE__*/React.createElement(Label, {
  inline: true,
  htmlFor: "checkbox2",
  ml: "default"
}, "Unchecked")), /*#__PURE__*/React.createElement(Box, {
  flexDirection: "column",
  marginRight: 15
}, /*#__PURE__*/React.createElement(Radio, {
  id: "checkbox3",
  checked: true,
  disabled: true
}), /*#__PURE__*/React.createElement(Label, {
  inline: true,
  htmlFor: "checkbox3",
  ml: "default"
}, "Disabled")));
const meta = {
  title: 'DesignSystem/Atoms/Radio',
  component: Radio
};
export default meta;
