// @adminjs/sequelize/lib/property.js
/* eslint-disable no-underscore-dangle */
import { BaseProperty } from 'adminjs';
const TYPES_MAPPING = [
    ['STRING', 'string'],
    ['TEXT', 'string'],
    ['INTEGER', 'number'],
    ['BIGINT', 'number'],
    ['FLOAT', 'float'],
    ['REAL', 'float'],
    ['DOUBLE', 'float'],
    ['DECIMAL', 'float'],
    ['DATE', 'datetime'],
    ['DATEONLY', 'date'],
    ['ENUM', 'string'],
    ['ARRAY', 'array'],
    ['JSON', 'object'],
    ['JSONB', 'object'],
    ['BLOB', 'string'],
    ['UUID', 'string'],
    ['CIDR', 'string'],
    ['INET', 'string'],
    ['MACADDR', 'string'],
    ['RANGE', 'string'],
    ['GEOMETRY', 'string'],
    ['BOOLEAN', 'boolean'],
];
class Property extends BaseProperty {
    sequelizePath;
    fieldName;
    constructor(sequelizePath) {
        const { fieldName } = sequelizePath;
        super({ path: fieldName });
        this.fieldName = fieldName;
        this.sequelizePath = sequelizePath;
    }
    name() {
        return this.fieldName;
    }
    isEditable() {
        if (this.sequelizePath._autoGenerated) {
            return false;
        }
        if (this.sequelizePath.autoIncrement) {
            return false;
        }
        if (this.isId()) {
            return false;
        }
        return true;
    }
    isVisible() {
        // fields containing password are hidden by default
        return !this.name().match('password');
    }
    isId() {
        return !!this.sequelizePath.primaryKey;
    }
    reference() {
        if (this.isArray()) {
            return null;
        }
        if (this.sequelizePath.references === 'string') {
            return this.sequelizePath.references;
        }
        if (this.sequelizePath.references && typeof this.sequelizePath.references !== 'string') {
            // Sequelize v7
            if (this.sequelizePath.references.table?.tableName) {
                return this.sequelizePath.references.table?.tableName;
            }
            // Sequelize v4+
            if (this.sequelizePath.references.model && typeof this.sequelizePath.references.model !== 'string') {
                return this.sequelizePath.references?.model?.tableName;
            }
            return this.sequelizePath.references?.model;
        }
        return null;
    }
    availableValues() {
        return this.sequelizePath.values && this.sequelizePath.values.length
            ? this.sequelizePath.values
            : null;
    }
    isArray() {
        return this.sequelizePath.type.constructor.name === 'ARRAY';
    }
    /**
     * @returns {PropertyType}
     */
    type() {
        let sequelizeType = this.sequelizePath.type;
        if (this.isArray()) {
            sequelizeType = sequelizeType.type;
        }
        const key = TYPES_MAPPING.find((element) => (sequelizeType.constructor.name === element[0]));
        if (this.reference()) {
            return 'reference';
        }
        const type = key && key[1];
        return (type || 'string');
    }
    isSortable() {
        return this.type() !== 'mixed' && !this.isArray();
    }
    isRequired() {
        return !(typeof this.sequelizePath.allowNull === 'undefined' || this.sequelizePath.allowNull);
    }
}
export default Property;
